src/Features/CSharp/Portable/MakeLocalFunctionStatic/GetCaptures.cs

```diff
@@ -0,0 +1,189 @@
+﻿
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeGeneration;
+using Microsoft.CodeAnalysis.CSharp.CodeGeneration;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.FindSymbols;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+
+
+
+namespace Microsoft.CodeAnalysis.CSharp.GetCapturedVariables
+{
+    [Shared]
+    [ExportLanguageService(typeof(GetCaptures), LanguageNames.CSharp)]
+    internal class GetCaptures : ILanguageService
+    {
+        private static readonly char[] s_underscore = { '_' };
+        private static readonly SyntaxGenerator s_generator = CSharpSyntaxGenerator.Instance;
+
+
+
+        internal async Task<Solution> CreateParameterSymbolAsync(Document document, LocalFunctionStatementSyntax localfunction, CancellationToken cancellationToken)
+        {
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(true);
+            if (semanticModel == null)
+            {
+                return document.Project.Solution;
+            }
+
+
+
+            var localFunctionSymbol = semanticModel.GetDeclaredSymbol(localfunction, cancellationToken);
+            var dataFlow = semanticModel.AnalyzeDataFlow(localfunction);
+            var captures = dataFlow.CapturedInside;
```

Comment:
We might want to copy the logic of retrieving captured locals into a separate method, so the consumer of this service can bail early if there's no captures (or contains certain captures we current don't support, e.g. `this`), instead of register a code action that makes no change to the code.

Reply:
Also, when you have done this, please change existing `MakeLocalFunctionStaticDiagnosticAnalyzer` to use this new method, which would reduce code duplication.